prover/src/lib.rs

```diff
@@ -333,8 +333,10 @@ pub trait Prover {
 
                 (None, AuxRandElements::new(rand_elements))
             };
+            let mut aux_trace = maybe_await!(self.build_aux_trace(&trace, &aux_rand_elements));
```

Comment:
Is there a reason to build the non-LogUp-GKR part of the aux trace here rather than before we build the GKR proof (i.e., do this on line 302 above).

If there is no particular reason, then I think it would be a bit more natural to first deal with non-LogUp-GKR part, then generate the GKR proof, and then build the LogUp-GKR-related columns.

Reply:
Yes, I agree. Have changed it now and simplified a few things on the way